src/main/java/edu/kit/compiler/JavaEasyCompiler.java

```diff
@@ -171,12 +175,30 @@ private static Result compileFirm(String filePath, Logger logger) {
             createAttributedAst(reader, logger, lexer, namespaceMapper);
 
             Firm.init("x86_64-linux-gnu", new String[]{ "pic=1" });
-            logger.info("Initialized libFirm Version: %s.%s\n",
+            logger.info("Initialized libFirm Version: %s.%s",
                 Firm.getMinorVersion(), Firm.getMajorVersion()
             );
 
-            // todo actually implement code generation
+            TypeMapper typeMapper = new TypeMapper(namespaceMapper, stringTable);
 
+            // todo insert Firm graph generation here
+
+            Lower.lower(typeMapper);
+
+            var sourceFile = new File(filePath).getName();
+            var assemblyFile = sourceFile + ".s";
+            logger.info("assembling program: '%s'", assemblyFile);
+            Backend.createAssembler(assemblyFile, sourceFile);
+
+            var stdLibrary = System.getenv("STD_LIBRARY_PATH");
+            if (stdLibrary == null || !new File(stdLibrary).exists()) {
+                logger.error("standard library implementation not found");
+                return Result.StandardLibraryError;
+            }
+
+            logger.info("compiling program: 'gcc \"%s\" \"%s\"'", assemblyFile, stdLibrary);
+            Runtime.getRuntime().exec(new String[]{ "gcc", assemblyFile, stdLibrary });
```

Comment:
Did you test whether this code works?

Reply:
Yes, I have have successfully compiled and executed code with this. The binary is written to the current working directory as `a.out`. This seems to be what _mjtest_ expects.

Reply:
:+1: